packages/material-ui/src/Popper/Popper.test.js

```diff
@@ -169,6 +167,37 @@ describe('<Popper />', () => {
     });
   });
 
+  describe('prop: keepMounted', () => {
+    it('should keep the children in the DOM', () => {
+      const wrapper = mount(<Popper {...defaultProps} open={false} keepMounted />);
+      assert.strictEqual(wrapper.find('span').length, 1);
+    });
+
+    it('should remove from DOM when closed on update frame immediately after mount', () => {
+      const wrapper = mount(<Popper {...defaultProps} open keepMounted={false} />);
+      wrapper.setProps({ open: false });
+      wrapper.update();
+      assert.strictEqual(wrapper.find('span').length, 0);
+    });
+
+    /* Test case for https://github.com/mui-org/material-ui/issues/15180 */
+    it('should remove from DOM when closed on update frame immediately after mount with transition prop', () => {
+      const wrapper = mount(
+        <Popper {...defaultProps} open={false} keepMounted={false} transition>
+          {({ TransitionProps }) => (
+            <Grow {...TransitionProps}>
+              <span>Hello World</span>
+            </Grow>
+          )}
+        </Popper>,
+      );
+      wrapper.setProps({ open: true });
+      wrapper.setProps({ open: false });
+      wrapper.update();
```

Comment:
Does this fail without it? I'm not sure injecting this here means that this is immediately asserting. It would help if you add some intermediate assertions so that it's obvious what's happening.

Reply:
Right you were about it being unnecessary. I've removed it now, and asserted that `setProps({ open: false })` is being called whilst the Popper still doesn't exist on the DOM (Due to the `Portal` double-rendering behaviour)

Reply:
I've also added a line-comment for future onlookers, because it's not immediately obvious _why_ the children don't exist on the DOM after the `setProps({ open: true })` call, unless you delve into `Popper` and `Portal`.